internal/serve/httphandler/mfa_handler.go

```diff
@@ -43,8 +44,36 @@ func (h MFAHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	deviceID := req.Header.Get(DeviceIDHeader)
+	if deviceID == "" {
+		httperror.BadRequest("Device-ID header is required", nil, nil).Render(rw)
+		return
+	}
+
+	if reqBody.MFACode == "" {
+		extras := map[string]interface{}{"mfa_code": "MFA Code is required"}
+		httperror.BadRequest("Request invalid", nil, extras).Render(rw)
+		return
+	}
+
+	user, err := h.AuthManager.GetUserByMFA(ctx, deviceID)
+	if err != nil {
+		if errors.Is(err, auth.ErrMFADeviceNotFound) {
+			// If we don't find the user by device id, we just return an ok response
+			// to prevent malicious client from searching for the device in the system
+			log.Ctx(ctx).Errorf("Device id in request not found: %s", deviceID)
+		}
+		httperror.InternalError(ctx, "Cannot get user by MFA", err, nil).Render(rw)
+		return
+	}
```

Comment:
🚨 having this piece of code before the recaptcha code causes the same security flaw as I raised last time on this same PR.

MFA is just 6 digits, it can be easily brute-forced if this return message is not placed **after** the recaptcha

Reply:
The naming for this method is misleading, so I've renamed it to `GetUserByDeviceID` per your suggestion below! I've also truncated the device ID in the log the same way we are doing for emails when searching for user by email in the other handlers